docs/src/main/asciidoc/amazon-lambda-guide.adoc

```diff
@@ -0,0 +1,258 @@
+
+== Update the Lambda
+
+You can update the Java code as you see fit.  Once you've rebuilt, you can redeploy your lambda by executing the
+`update.sh` script.
+
+[source, subs=attributes+]
+----
+aws lambda update-function-code --function-name my-function \
+           --zip-file fileb://target/my-function-1.0-SNAPSHOT-runner.jar
+----
+
+== Deploy to AWS Lambda Custom (native) Runtime
+
+If you want a lower memory footprint and faster initialization times for your lambda, you can compile your Java
+code to a native executable using Graal.  Just make sure to rebuild your project with the `-Dnative` switch.
+
+== Build and Deploy (native)
+
+Build the native executable.
+
+[source, subs=attributes+]
+----
+mvn clean install -Dnative
+----
+
+This will compile and create a native executable image using Graal.  It also generates a zip file `target/function.zip`.
+This zip file contains your native executable image renamed to `bootstrap`.  This is a requirement of Amazon Lambda
+Custom Runtime.
+
+
+== Create an Execution Role
+
+View the https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-awscli.html[Getting Started Guide] for deploying
+a lambda with AWS CLI.  Specifically make sure you have created an `Execution Role`.  You will need the `Role Arn` to copy/paste
+into the scripts generated by the maven archetype.
+
+== Edit native script files
+
+Edit the ``create-native.sh` scripts included with the generated project.  After the `--role` switch
+replace the dummy `Role Arn` with the Arn of the `Execution Role` you created in the Amazon IAM console.
+
+== Create the Native Lambda
+
+The `create-native.sh` script is for deploying your lambda using the AWS Lambda Custom runtime.
+
+[source, subs=attributes+]
+----
+aws lambda create-function --function-name my-native-function \
+           --zip-file fileb://target/function.zip --handler any.name.not.used \
+           --runtime provided --role arn:aws:iam::1234567:role/lambda-cli-role \
+           --environment Variables="{DISABLE_SIGNAL_HANDLERS=true}"
+----
+
+Don't forget to replace the `--role` flag with the Arn of your `Execution Role`.
+
+Notice that the `--zip-file` flag points to the `function.zip` file generated by the build.
+The handler switch is not used, but must be set.  It can be any value.  Finally, notice that an environment variable
+must be set: `DISABLE_SIGNAL_HANDLERS`.  This resolves an incompatibility with the Amazon Lambda Custom Runtime container
+and Graal.
+
+If there are any problems creating the function, you must delete it with the `delete-native.sh` script before re-running
+the `create-native.sh` switch.
+
+== Invoke the Native Lambda
+
+Use the `invoke-native.sh` script to invoke your function.
+
+[source, subs=attributes+]
+----
+aws lambda invoke --function-name my-native-function
+                  --payload file://payload.json out
+                  --log-type Tail --query 'LogResult' --output text |  base64 -d
+----
```

Comment:
I'm a bit confused here. You say Use the `invoke-native.sh` and then you show something that does not use it. Or is it the content of the script file?

Reply:
its the content of the script file

Reply:
I put a title for each script section.  That should clarify